### ISSUE TITLE
Run on GPU?

### ISSUE THREAD

**Issue by cjmielke** [jacobgil/keras-grad-cam#4]
Thank you for this immensely awesome demo. I think this technique opens many doors.

It took me a lot of fiddling to get reasonable output, since my network architecture is quite a bit different. All is good now. However I noticed that things are running on the CPU. In keras, my other scripts are working on the GPU with the tensorflow backend.

Would you happen to know offhand how to ensure the K.function and K.gradients calls in here run on the GPU. Im digging into the keras source to understand that they are just frontends for tensorflow functions, but I get a bit lost here.

Thanks!

**Comment by jacobgil** [comment]
Hi,
In case of tensorflow, K.gradients uses tf.gradients().
If you are using the tensorflow backend compiled with CUDA support, it should run on the GPU.

How are you checking if it runs on CPU/GPU ? 
When running the exact code in this repository, does it still run on the CPU?

**Comment by cjmielke** [comment]
I think I figured my issue out. My slowness was primarily brought about by looping over many images. Speeding it up was just a matter of hoisting the function declaration out of the loop.

On my end, its still CPU bound, but that's alright in my case.

Thanks! Great work. I'm digging into the literature to see if the other half of the grad cam algorithm can be implemented with Keras.
